packages/core/src/storage/compare.ts

```diff
@@ -79,10 +79,15 @@ export class StorageLayoutComparator {
   }
 
   getFieldChange<F extends StorageField>(original: F, updated: F): StorageFieldChange<F> | undefined {
-    const nameChange = !this.unsafeAllowRenames && original.label !== updated.label;
-    const typeChange = this.getTypeChange(original.type, updated.type, { allowAppend: false });
+    const nameChange = !this.unsafeAllowRenames && original.label !== (updated.rename ?? updated.label);
+    let type: ParsedTypeDetailed | undefined = undefined;
+    if (original.type.item.label == updated.retyped?.trim()) {
+      type = original.type;
+    }
+    const typeChange = this.getTypeChange(original.type, type ?? updated.type, { allowAppend: false });
```

Comment:
It looks like we are not verifying that the retype is valid.

If I go from
```
uint64 a;
uint64 b;
uint64 c;
```
to
```
uint64 a;
/// @custom:oz-retyped-from uint64
uint256 b;
uint64 c;
```

This will be accepted.

IMO it shouldn't. I believe we should try to notify that the retype is dangerous.
Then we could (also) have a `oz-force-retyped-from uint64` if they really want to shoot them in the foot.

Reply:
The `getLayoutChange` is where it checks that the retyped is valid, happens after this validation